### PR TITLE
[stable/redmine] Add option to use existingSecret for external db

### PR DESCRIPTION
**Pull request by crmarques**
#### What this PR does / why we need it:

Add option to use existingSecret for external db

#### Special notes for your reviewer:

This PR does not change chart's default behavior.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by crmarques** [comment]
@carrodher, could you, please, review this addendum to #19011 ?

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @crmarques 

Thanks for the PR!!!

What about using the existing parameter `existingsecret` to provide these credentials?

Currently this parameter is ignored when using an external database, but we could change the logic so we use this parameter when it's set (both when using an external database or using a MariaDB/PostgreSQL subchart)

**Comment by carrodher** [comment]
Thanks for the contribution, as it is new functionality, can you increment the chart version in a minor? According to [semver](http://semver.org/):

> Given a version number `MAJOR.MINOR.PATCH`, increment the:
> - MAJOR version when you make incompatible API changes,
> - MINOR version when you add functionality in a backwards-compatible manner, and
> - PATCH version when you make backwards-compatible bug fixes.
> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

**Comment by crmarques** [comment]
@carrodher, version bumped.

**Comment by crmarques** [comment]
@juan131, I think this would be a good solution but it would break compatibility. Could we proceed with this minor change, proposed by this PR, and work on this approach in the next version, with a major version?

**Comment by juan131** [comment]
Hi @crmarques 

> I think this would be a good solution but it would break compatibility.

Why? Could you please detail why you think it'll brea backwards compatibility? I would like to understand the implications.

**Comment by crmarques** [comment]
@juan131, I think I misunderstood you first comment. 

I've just changed the chart's behavior for using only one secret, with option for user to indicate an `existingSecret`, with key `external-db-password`. Could you review the changes?

File: stable/redmine/README.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ The following table lists the configurable parameters of the Redmine chart and t
 | `smtpUser`                          | SMTP user                                  | `nil`                                                   |
 | `smtpPassword`                      | SMTP password                              | `nil`                                                   |
 | `smtpTls`                           | Use TLS encryption with SMTP               | `nil`                                                   |
-| `existingSecret`                    | Use existing secret for password details (`smtpPassword` and `redminePassword` will be ignored and picked up from this secret). The secret has to contain the keys `redmine-password` and `smtp-password`. | `nil`                        |
+| `existingSecret`                    | Use existing secret for password details (`redminePassword`, `smtpPassword` and `externalDatabase.password` will be ignored and picked up from this secret). The secret must contain the keys `redmine-password`, `smtp-password` and `external-db-password`. | `nil`                        |
```
Comment (juan131) [inline stable/redmine/README.md]: This is not 100% true:

```
The secret must contain the keys `redmine-password`, `smtp-password` and `external-db-password`.
```

```
It must contain the keys `redmine-password` and `smtp-password` when `postgresql.enabled=true` or `mariadb.enabled=true`
It must contain the key `external-db-password` when `postgresql.enabled=false` and `mariadb.enabled=false`

File: stable/redmine/templates/secrets.yaml [hunk 2]
```diff
@@ -15,5 +15,10 @@ data:
   {{ else }}
   redmine-password: {{ randAlphaNum 10 | b64enc | quote }}
   {{ end }}
+  {{ if .Values.externalDatabase.password }}
+  external-db-password: {{ default "" .Values.externalDatabase.password | b64enc | quote }}
+  {{ else }}
+  external-db-password: {{ randAlphaNum 10 | b64enc | quote }}
+  {{ end }}
```
Comment (juan131) [inline stable/redmine/templates/secrets.yaml]: Suggestion:
```diff
+ {{- if or (and .Values.databaseType.mariadb (not .Values.mariadb.enabled)) (and .Values.databaseType.postgresql (not .Values.postgresql.enabled)) }}
   {{- if .Values.externalDatabase.password }}
   external-db-password: {{ default "" .Values.externalDatabase.password | b64enc | quote }}
   {{- else }}
   external-db-password: {{ randAlphaNum 10 | b64enc | quote }}
   {{- end }}
+  {{- end }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the changes! Please take a look to a couple of suggestions.

File: stable/redmine/README.md (hunk 1 above)
Comment (crmarques) [reply stable/redmine/README.md]: Fixed README.md

File: stable/redmine/templates/secrets.yaml (hunk 2 above)
Comment (crmarques) [reply stable/redmine/templates/secrets.yaml]: You are correct. Added if statement for "external-db-password".

**Comment by crmarques** [comment]
@juan131, I included your suggestions. Anything else?

**Comment by crmarques** [comment]
@juan131, can we proceed with this PR?

**Review by juan131 — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**